nemo/collections/nlp/models/duplex_text_normalization/duplex_decoder.py

```diff
@@ -50,6 +51,27 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         # Language
         self.lang = cfg.get('lang', None)
 
+        # Covering Grammars
+        self.cg_normalizer = None  # Default
+        # We only support integrating with English TN covering grammars at the moment
+        self.use_cg = cfg.get('use_cg', False) and self.lang == constants.ENGLISH
+        if self.use_cg:
+            self.setup_cgs(cfg)
+
+    # Setup covering grammars (if enabled)
+    def setup_cgs(self, cfg: DictConfig):
+        """
+        Setup covering grammars (if enabled).
+        :param cfg: Configs of the decoder model.
+        """
+        self.use_cg = True
+        self.neural_confidence_threshold = cfg.get('neural_confidence_threshold', 0.99)
+        self.n_tagged = cfg.get('n_tagged', 1)
+        input_case = 'cased'  # input_case is cased by default
+        if hasattr(self._tokenizer, 'do_lower_case') and self._tokenizer.do_lower_case:
+            input_case = 'lower_cased'
+        self.cg_normalizer = NormalizerWithAudio(input_case=input_case, lang=self.lang)
```

Comment:
add a PYNINI_AVAILABLE check like [here](https://github.com/NVIDIA/NeMo/blob/main/tests/nemo_text_processing/en/test_address.py#L25)

if it's not available and use_cg==True -> raise an error

Reply:
Yes. I added that.